### PR TITLE
Start of the python interface

### PR DESCRIPTION
**Pull request by maartenflippo**
This PR starts the introduction of the python interface. Since it is a big addition, I broke it up into smaller amounts for review. This PR sets up the project and implements the bare minimum to run an nqueens model from python (see the included sample).

There are still some remnants from the project scaffold. In particular the `tests` directory in `pumpkin-py`. However, I do not yet know how to test the python interface beyond duplicating many of the tests we already have. I have started an issue #34 to discuss testing this code, so we don't have to focus on that in this PR.

### CODE REVIEW

File: pumpkin-cli/src/flatzinc/compiler/post_constraints.rs [hunk 1]
```diff
@@ -256,7 +256,7 @@ fn compile_array_int_maximum(
     let rhs = context.resolve_integer_variable(&exprs[0])?;
     let array = context.resolve_integer_variable_array(&exprs[1])?;
 
-    Ok(constraints::maximum(array.as_ref(), rhs)
+    Ok(constraints::maximum(array.as_ref().to_owned(), rhs)
```
Comment (ImkoMarijnissen) [inline pumpkin-cli/src/flatzinc/compiler/post_constraints.rs:259]: Why this `to_owned`?

File: pumpkin-py/python/tests/test_all.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import pytest
+import pumpkin_py
+
+
+def test_sum_as_string():
```
Comment (ImkoMarijnissen) [inline pumpkin-py/python/tests/test_all.py:5]: Maybe just remove?

**Review by ImkoMarijnissen — CHANGES_REQUESTED** [review]

File: pumpkin-cli/src/flatzinc/compiler/post_constraints.rs (hunk 1 above)
Comment (maartenflippo) [reply pumpkin-cli/src/flatzinc/compiler/post_constraints.rs:259]: As discussed, this has to do with the lifetimes passing through constraint poster, and rust not being able to statically guarantee that the array will be alive when the constraint is posted.

**Review by ImkoMarijnissen — DISMISSED** [review]
Can we just create an enum in the python part rather than these additional methods on the Constraint interface?

File: pumpkin-lib/src/propagators/arithmetic/absolute_value.rs [hunk 3]
```diff
@@ -12,7 +12,7 @@ use crate::engine::variables::IntegerVariable;
 ///
 /// The propagator is bounds consistent wrt signed. That means that if `signed \in {-2, -1, 1, 2}`,
 /// the propagator will not propagate `[absolute >= 1]`.
-#[derive(Debug)]
+#[derive(Clone, Debug)]
```
Comment (ImkoMarijnissen) [inline pumpkin-lib/src/propagators/arithmetic/absolute_value.rs:15]: Would it be possible to remove the clone from the constraints now?

**Review by ImkoMarijnissen — DISMISSED** [review]

**Review by ImkoMarijnissen — APPROVED** [review]

**Pull request merged**